django_lti_tool_provider/tests/test_views.py

```diff
@@ -133,36 +141,20 @@ def test_no_session_given_incorrect_payload_throws_bad_request(self, _):
         self.assertIn("Invalid LTI Request", response.content)
 
     def test_no_session_correct_payload_processes_lti_request(self, patched_send_lti_received):
-        with self.assertRaises(LtiUserData.DoesNotExist):
-            LtiUserData.objects.get(user=self.user)  # precondition check
+        # Precondition check
+        self.assertFalse(LtiUserData.objects.all())
 
         response = self.send_lti_request(self.get_correct_lti_payload())
 
-        self._verify_lti_created(self.user, self._data)
-        self._verify_redirected_to(response, self._url_base + self.DEFAULT_REDIRECT)
-        self._verify_lti_updated_signal_is_sent(patched_send_lti_received, self.user)
-
-    @ddt.data('GET', 'POST')
-    def test_session_set_processes_lti_request(self, method, patched_send_lti_received):
```

Comment:
@Kelketek true, but only if authentication hook is (1) provided and (2) *always* authenticates users. Since it intended to be a generic tool not tied to dalite only, it is possible that one or both of this assumptions is not true.

The scenario I have in mind is a bit complicated, but viable (especially if app actually manages users, i.e. allows creating logins and logging in not only via LTI). Basically, first LTI might not be authenticated -> user redirected to auth backend (e.g. django.contrib.auth view) -> decides that he forgot password or something and not proceeds with auth immediately; closes the window (but session lti data is already set). Than he logs in normally (not via LTI request), after that it sends new LTI request. THis test ensures that in such scenario second LTI request prevails.

So, it's not the case with dalite, but it is a useful test; please restore it :)